docs/concepts/how-ipfs-works.md

```diff
@@ -5,86 +5,171 @@ description: Learn how the InterPlanetary File System (IPFS) works and why it's
+### mDNS
+
+To quickly and efficiently discover peers, IPFS uses Multicast Domain Name System (mDNS), a type of <VueCustomTooltip label="A system in which human-readable internet domain names are mapped to IP addresses. The Domain Name System has been an essential component of the internet since 1985." underlined multiline is-medium>DNS</VueCustomTooltip> protocol that resolves human-readable internet domain names to IP names without the use of a <VueCustomTooltip label="Any computer application that implements a system in which human-readable internet domain names are mapped to IP addresses (DNS)." underlined multiline is-medium>name server</VueCustomTooltip>. The process for peer discovery is as follows:
+
+1. Given a peer ID, a node broadcasts a query message to multiple peers in the network, asking that the peer having the given peer ID to identify itself.
+2. The peer with the given peer ID then broadcasts a message with its IP address to other peers on the network, which can then use that information to update their own mDNS records.
+
+The use of mDNS in IPFS has several benefits that make it a clear choice for peer discovery over traditional DNS:
+
+- Nodes can broadcast messages requesting and reporting address information to multiple peers, increasing speed and efficiency of peer discovery.
+- Nodes do not rely on a centralized name server to discover other peers, improving network resilience, decentralization and censorship resistance.
+
+### Delegated routing over HTTP
 
-Another useful feature of Merkle DAGs and breaking content into blocks is that if you have two similar files, they can share parts of the Merkle DAG, i.e., parts of different Merkle DAGs can reference the same subset of data. For example, if you update a website, only updated files receive new content addresses. Your old version and your new version can refer to the same blocks for everything else. This can make transferring versions of large datasets (such as genomics research or weather data) more efficient because you only need to transfer the parts that are new or changed, instead of creating entirely new files each time.
+Because IPFS is an open-source protocol with multiple implementations, an IPFS node is not required to implement full routing functionality itself. Instead, an IPFS node can request that a <VueCustomTooltip label="An IPFS node that perform tasks on behalf of network peers using the IPFS HTTP API." underlined multiline is-medium>delegated router</VueCustomTooltip> perform a routing task for it. For example, if an IPFS node does not implement the DHT, a delegated router can search the DHT for peers on its behalf. The main benefit of delegated routing is that nodes are not required to implement routing functionality themselves if they do not wish to, or do not have the computing resources to do so.
```

Comment:
```suggestion
Delegated content routing is a mechanism for IPFS implementations to use for offloading content routing to another process/server using an HTTP API. For example, if an IPFS node does not implement the DHT, a delegated router can search the DHT for peers on its behalf. The main benefit of delegated routing is that nodes are not required to implement routing functionality themselves if they do not wish to, or do not have the computing resources to do so.
```

I think the opening sentence here misses the point. "Full routing functionality" for IPFS has never really been defined, so I would avoid referencing it here. 

Another thing to note about delegating routing is that it's not just a way to delegate DHT work, as is the case of the network indexer. It's a standard interface that allows more flexibility in terms of how content routing works. It can be used for example, for private data sets or to build an IPFS system that doesn't use the DHT and instead has a custom backend that uses Redis and exposes a delegated routing endpoint.